pynest/nest/lib/hl_api_connections.py

```diff
@@ -0,0 +1,641 @@
+
+        for s, p in zip(pre, params):
+            sps(s)
+            sps(p)
+            sr(cmd)
+    else:
+        # Call the variant where all connections are given explicitly
+
+        # Disable dict checking, because most models can't re-use their own status dict
+        dict_miss = GetKernelStatus('dict_miss_is_error')
+        SetKernelStatus({'dict_miss_is_error': False})
+
+        sps(pre)
+        sr('DataConnect_a')
+
+        SetKernelStatus({'dict_miss_is_error': dict_miss})
+
+
+@check_stack
+@deprecated(alt_func_name='Connect')
+def RandomDivergentConnect(pre, post, n, weight=None, delay=None, model="static_synapse", options=None):
+    """
+    Connect each neuron in pre to n randomly selected neurons from
+    post. pre and post have to be lists. If weight is given (as a
+    single float or as list of floats), delay also has to be given as
+    float or as list of floats. options is a dictionary specifying
+    options to the RandomDivergentConnect function: allow_autapses,
+    allow_multapses.
+    """
+
+    if not isinstance(n, int):
+        raise TypeError("number of neurons n should be an integer")
+
+    # store current options, set desired options
+    old_options = None
+    error = False
+    if options is not None:
+        old_options = sli_func('GetOptions', '/RandomDivergentConnect',
+                               litconv=True)
+        del old_options['DefaultOptions'] # in the way when restoring
+        sli_func('SetOptions', '/RandomDivergentConnect', options,
+                 litconv=True)
+
+    if weight is None and delay is None:
+        sli_func(
+            '/m Set /n Set /post Set { n post m RandomDivergentConnect } forall',
+            pre, post, n, '/'+model, litconv=True)
+
+    elif weight is not None and delay is not None:
+        weight = broadcast(weight, n, (float,), "weight")
+        if len(weight) != n:
+            raise kernel.NESTError("weight must be a float, or sequence of floats of length 1 or n")
+        delay = broadcast(delay, n, (float,), "delay")
+        if len(delay) != n:
+            raise kernel.NESTError("delay must be a float, or sequence of floats of length 1 or n")
+
+        sli_func(
+            '/m Set /d Set /w Set /n Set /post Set { n post w d m RandomDivergentConnect } forall',
+            pre, post, n, weight, delay, '/'+model, litconv=True)
+
+    else:
+        error = True
+
+    # restore old options
+    if old_options is not None:
+        sli_func('SetOptions', '/RandomDivergentConnect', old_options,
+                 litconv=True)
+
+    if error:
+        raise kernel.NESTError("Both 'weight' and 'delay' have to be given.")
+
+def _is_subnet_instance(gids):
+    "Returns true if all gids point to subnet or derived type."
+
+    try:
+        GetChildren(gids)
+        return True
+    except kernel.NESTError:
+        return False
+
+@check_stack
+def CGConnect(pre, post, cg, parameter_map=None, model="static_synapse"):
+    """
```

Comment:
@tammoippen: technically I agree. I just thought that this PR should just split the HL API as it is now and further splitting can be done in our refactoring branch. The automatic module discovery implemented in `__init__.py` lines 179-181 should make it fairly simple to have modules install their own files if they want to.